### PR TITLE
O21 kb 9

### PR DESCRIPTION
**Pull request by shaynelar**
Can delete warehouses form the ui via a modal now.
Made some changes to code on the  warehouse page and warehouse items pages as the modal is triggered and rendered within those pages

### CODE REVIEW

File: client/src/components/DeleteModal/DeleteModal.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	async function deleteWarehouse() {
+		try {
+			await axios.delete(`${API_URL}/warehouses/${deleteId}`);
+			closeModal();
+		} catch (err) {
+			//noop
+		}
+	}
+	return (
+		<>
```
Comment (manecipoglu) [inline client/src/components/DeleteModal/DeleteModal.jsx:22]: Looks like fragments here can be removed

File: client/src/components/DeleteModal/DeleteModal.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+			{isOpen && (
```
Comment (manecipoglu) [inline client/src/components/DeleteModal/DeleteModal.jsx:23]: Do we need to check if the modal is open? The isOpen boolean comes from props and every time the component is rendered it should be open. Unless of course we render this component all the time. If that's the case, conditional rendering from the parent may be another option to consider.

File: client/src/components/DeleteModal/DeleteModal.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+				<div className="modal__background">
+					<form className="modal__body">
```
Comment (manecipoglu) [inline client/src/components/DeleteModal/DeleteModal.jsx:25]: I am not sure if a form element is the best semantic here. It won't change the functionality of course.

File: server/data/warehouses.json [hunk 4]
```diff
@@ -1,30 +1,4 @@
...
-    "address": "469 King Street West",
-    "city": "Toronto",
-    "country": "CAN",
-    "contact": {
-      "name": "Greame Lyon",
-      "position": "Warehouse Manager",
-      "phone": "+1 (646) 123-1234",
-      "email": "glyon@instock.com"
-    }
-  },
```
Comment (manecipoglu) [inline server/data/warehouses.json:27]: Please bring back our deleted warehouse :)))

**Comment by manecipoglu** [comment]
Thank you for the modal PR Shayne. It's not easy to implement modals manually. One question, after we delete the warehouse and the modal is closed, is it reflected on the list of warehouses?

File: server/data/warehouses.json (hunk 4 above)
Comment (shaynelar) [reply server/data/warehouses.json:27]: well have to reset the forms to initial data for submission

File: client/src/components/DeleteModal/DeleteModal.jsx (hunk 1 above)
Comment (shaynelar) [reply client/src/components/DeleteModal/DeleteModal.jsx:22]: You actually need a fragment in order to use javascript inside {}  in the return statement. curly braces cant be top level

File: client/src/components/DeleteModal/DeleteModal.jsx (hunk 3 above)
Comment (shaynelar) [reply client/src/components/DeleteModal/DeleteModal.jsx:25]: yeah wasn't sure. I was thinking it was kind of a form but can probably use something else instead

**Comment by shaynelar** [comment]
@manecipoglu yes the ui will remove the deleted warehouse

File: client/src/components/DeleteModal/DeleteModal.jsx (hunk 2 above)
Comment (shaynelar) [reply client/src/components/DeleteModal/DeleteModal.jsx:23]: we could yes, but currently it is rendered all the time in warehouse and I have the modal controlling its visibility via a prop

File: client/src/components/DeleteModal/DeleteModal.jsx (hunk 1 above)
Comment (manecipoglu) [reply client/src/components/DeleteModal/DeleteModal.jsx:22]: You can actually omit the curly braces too if you remove the frangment and just go isOpen && ...

Comment (shaynelar) [reply client/src/components/DeleteModal/DeleteModal.jsx:22]: cool didn't know that!

File: client/src/components/WareHouseItem/WareHouseItem.jsx [hunk 5]
```diff
@@ -2,44 +2,45 @@ import { Link } from "react-router-dom";
...
+				</div>
+				<div className="warehouse__container warehouse__container--contact">
+					<p className="warehouse__title">CONTACT INFORMATION</p>
+					<address className="warehouse__copy">
+						<p>{phone}</p>
+						<p>{email}</p>
+					</address>
+				</div>
+				<div className="warehouse__icons">
+					<i onClick={(e) => handleModal(e, name, id)} className="icon icon--delete" />
```
Comment (shaynelar) [inline client/src/components/WareHouseItem/WareHouseItem.jsx:40]: this is not really semantically correct but the styling messed up when I changed it from a Link aka <a /> tag so I can refactor this today

Comment (manecipoglu) [reply client/src/components/WareHouseItem/WareHouseItem.jsx:40]: I believe for icons using a `<span>` is semantically better than `<i>` if you are going to refactor anyway but no difference in functionality.

**Comment by mashakeita** [comment]
Everything looks good

**Pull request merged**